server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -624,6 +624,7 @@ protected Node(
                 additionalSettingsFilter,
                 settingsUpgraders
             );
+            threadPool.setClusterSettings(settingsModule.getClusterSettings());
```

Comment:
Is there any way we can avoid the cyclic dependency of these 2 classes one each other?